lua/telekasten.lua

```diff
@@ -201,34 +201,44 @@ local function check_dir_and_ask(dir, purpose)
                     vim.cmd('echomsg " "')
                     vim.cmd('echomsg "' .. dir .. ' created"')
                     ret = true
+                    callback(ret)
                 else
                     -- unreachable: plenary.Path:mkdir() will error out
                     tkutils.print_error("Could not create directory " .. dir)
                     ret = false
+                    callback(ret)
                 end
             end
         end)
     else
         ret = true
+        if callback ~= nil then
+            callback(ret)
+        end
+        return ret
     end
-    return ret
 end
 
-local function global_dir_check()
+local function global_dir_check(callback)
```

Comment:
this is one of the main functions affected by the refactor of `check_dir_and_ask`, therefore all functions that call `global_dir_check` are affected and need to pass in a callback